### PR TITLE
chore: update to go1.19

### PR DESCRIPTION
**Pull request by ldez**
- updates CI to use go1.19 and go1.20
- uses either `URL.JoinPath` or `url.JoinPath` instead of `url.Parse`+ `path.Join` => removes more than 250 lines!
- update linter

https://tip.golang.org/doc/go1.19

Note:
`URL.JoinPath` (or `url.JoinPath`) is not the exact equivalent of `url.Parse`+ `path.Join` because the path elements in `URL.JoinPath` are path encoded which means that if an element contains a query string the query string will be path encoded (`%3F` instead of `?`).

### CODE REVIEW

**Comment by dmke** [comment]
Will take a look later.

File: .github/workflows/go-cross.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
 
     strategy:
       matrix:
-        go-version: [ 1.18, 1.19, 1.x ]
+        go-version: [ 1.19, '1.20', 1.x ]
```
Comment (dmke) [inline .github/workflows/go-cross.yml:19]: Let's generally quote things that could be mistaken as floating point numbers by YAML parsers:

```suggestion
        go-version: [ '1.19', '1.20', 1.x ]
```

File: providers/dns/hyperone/internal/client.go [hunk 2]
```diff
@@ -77,9 +76,12 @@ func NewClient(apiEndpoint, locationID string, passport *Passport) (*Client, err
 // https://api.hyperone.com/v2/docs#operation/dns_project_zone_recordset_list
 func (c *Client) FindRecordset(zoneID, recordType, name string) (*Recordset, error) {
 	// https://api.hyperone.com/v2/dns/{locationId}/project/{projectId}/zone/{zoneId}/recordset
-	resourceURL := path.Join("dns", c.locationID, "project", c.projectID, "zone", zoneID, "recordset")
+	endpoint, err := url.JoinPath(c.apiEndpoint, "dns", c.locationID, "project", c.projectID, "zone", zoneID, "recordset")
+	if err != nil {
+		return nil, err
+	}
```
Comment (dmke) [inline providers/dns/hyperone/internal/client.go:82]: There's a lot of repetition here, in `CreateRecordset`, in `DeleteRecordset`, in `GetRecords`, and in `GetZones`.

Can we make `(*Client).apiEndpoint` a `*url.URL`, and use `c.apiEndoint.JoinPath(...)`?

**Review by dmke — CHANGES_REQUESTED** [review]
Nice cleanup, `JoinPath` makes this much more readable than `path.Join`+`url.Parse`+error handling.

In the hyperone client, I'd like to see the URL parsing moved to `NewClient`. This also has the benefit of catching parse errors at initialization time, and not at a later `Present`/`Cleanup` call.

File: providers/dns/hyperone/internal/client.go (hunk 2 above)
Comment (ldez) [reply providers/dns/hyperone/internal/client.go]: something like that? https://github.com/go-acme/lego/pull/1835/commits/b0b91728c49a6cb396fbfe8729db5ea474350e3e

Comment (dmke) [reply providers/dns/hyperone/internal/client.go]: That, plus

```diff
-endpoint, err := url.JoinPath(c.apiEndpoint, "dns", c.locationID, "project", c.projectID, "zone", zoneID, "recordset")
-if err != nil {
-	return nil, err
-}
+endpoint := c.apiEndpoint.JoinPath("dns", c.locationID, "project", c.projectID, "zone", zoneID, "recordset")
```

Comment (ldez) [reply providers/dns/hyperone/internal/client.go]: is it better like that? https://github.com/go-acme/lego/pull/1835/commits/5fd749110ab553bda056b1b81a4991c6e599c907

Comment (ldez) [reply providers/dns/hyperone/internal/client.go]: clean the constructor in https://github.com/go-acme/lego/pull/1835/commits/0d78446553de4c048ab3366cb1d8a4d9cc1b96fb

**Pull request merged**